### PR TITLE
feat(blocks): Add support for additional LLM providers to LLM Block

### PR DESCRIPTION
**Pull request by Torantulino**
This commit adds support for the following models:

```python
# OpenAI Models
GPT4O = "gpt-4o"
GPT4_TURBO = "gpt-4-turbo"
GPT3_5_TURBO = "gpt-3.5-turbo"

# Anthropic models
CLAUDE_3_5_SONNET = "claude-3-5-sonnet-20240620"
CLAUDE_3_HAIKU = "claude-3-haiku-20240307"

# Groq models
LLAMA3_8B = "llama3-8b-8192"
LLAMA3_70B = "llama3-70b-8192"
MIXTRAL_8X7B = "mixtral-8x7b-32768"
GEMMA_7B = "gemma-7b-it"
GEMMA2_9B = "gemma2-9b-it"
```

Every model has been tested with a single LLM block and is confirmed to be working in that setup.

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/blocks/ai.py [hunk 1]
```diff
@@ -1,25 +1,59 @@
...
+    LlmModel.MIXTRAL_8X7B: ModelMetadata("groq", 32768),
+    LlmModel.GEMMA_7B: ModelMetadata("groq", 8192),
+    LlmModel.GEMMA2_9B: ModelMetadata("groq", 8192),
+}
 
 class LlmCallBlock(Block):
     class Input(BlockSchema):
         prompt: str
-        api_key: BlockFieldSecret = BlockFieldSecret(key="openai_api_key")
+        api_key: BlockFieldSecret
```
Comment (Torantulino) [inline rnd/autogpt_server/autogpt_server/blocks/ai.py]: @majdyz can you take a look at this? I may have broken server-side secrets

**Comment by Torantulino** [comment]
Just found an Error with Anthropic models:
```
rror calling LLM: Error code: 400 - {'type': 'error', 'error': {'type': 'invalid_request_error', 'message': 'messages: Unexpected role "system". The Messages API accepts a top-level `system` parameter, not "system" as an input message role.'}}
```

File: rnd/autogpt_server/autogpt_server/blocks/ai.py [hunk 2]
```diff
@@ -95,34 +148,38 @@ def parse_response(resp: str) -> tuple[dict[str, str], str | None]:
...
+                  |And this is the error:
+                  |--
+                  |{parsed_error}
+                  |--
+                """
+                prompt.append({"role": "user", "content": trim_prompt(retry_prompt)})
+            except Exception as e:
+                logger.error(f"Error calling LLM: {e}")
+                retry_prompt = f"Error calling LLM: {e}"
+
```
Comment (majdyz) [inline rnd/autogpt_server/autogpt_server/blocks/ai.py:200]: new line ?

File: rnd/autogpt_server/autogpt_server/blocks/ai.py (hunk 1 above)
Comment (majdyz) [reply rnd/autogpt_server/autogpt_server/blocks/ai.py]: This makes api_key becomes a mandatory input field. 
Can we change the key to something more generic? e.g: LLM_API_KEY and use just that ?

**Review by majdyz — APPROVED** [review]

**Review by majdyz — APPROVED** [review]

**Review by majdyz — DISMISSED** [review]

**Review by majdyz — APPROVED** [review]

**Pull request merged**